_Week8/group1/simulate.do

```diff
@@ -0,0 +1,116 @@
+cap program drop salary_simulate // note: did not name program 'simulate' because simulate is a pre-existing Stata command
+
+program define salary_simulate, rclass // define an r-class program so that the program writes into return and we can get stuff out of the program
+	syntax anything
+
+	// clear data in memory
+	clear
+
+	// generate blank data (observations without any content)
+	set obs `anything' // `anything' is number of states sampled from
+	
+	* ---------- generate clusters and associated fixed effects ----------*
+	// generate state cluster
+	generate state = _n // generate state clusters
+	label variable state "State in which college or university is located"
+	
+	generate u_i = rchi2(4)*1000 // generate fixed effects for states
+	generate random1 = runiform() // generate random numbers to be used to vary number of schools per region
+	
+	// generate university clusters within states, with 15-25 universities per state 
+	expand 15 if random1 < 0.3
+	expand 20 if random1 >= 0.3 & random1 < 0.7
+	expand 25 if random1 >= 0.7
+	bysort state: generate university = _n // generate university clusters
+	label variable university "College or university that student attended"
+	
+	generate e_ij = rnormal(0,5)*100 // generate fixed effects for universities
+	generate random2 = runiform()  // generate random numbers to be used to vary number of students per school
+	
+	// generate student observations, with 500-1000 students per university
```

Comment:
There was no reason for the precise numbers other than we wanted to randomize it a bit, so they were arbitrarily selected. Good idea to randomize it even further though, I'll make a note to edit. Could you clarify what you mean by DGPs?

Reply:
Got it. DGP = data generating process. I meant that you could apply the procedure I explained to a few of the instances in which you randomly generate data. Please let me know if you have further questions/clarifications.

Reply:
Note that `expand` can take a variable as an argument instead of a fixed value, which can be very useful! That is, by using `trunc` or `round` as you have in other places you can very quickly create a distribution of sizes!

Reply:
@f2pHgty8hw: Yes, I realized what you meant during class today! Thanks.

@bbdaniels: Oh got it! I was wondering how we could have used `expand` here... during editing, I'll see where we can use that command.